contracts/contracts/vault/VaultCore.sol

```diff
@@ -359,31 +359,39 @@ contract VaultCore is VaultStorage {
      *      strategies and update the supply of OUSD.
      * @return uint256 New total supply of OUSD
      */
-    function rebase()
-        public
-        whenNotRebasePaused
-        nonReentrant
-        returns (uint256 newTotalSupply)
-    {
-        return _rebase();
+    function rebase() public whenNotRebasePaused nonReentrant {
+        _rebase();
     }
 
     /**
      * @dev Calculate the total value of assets held by the Vault and all
-     *      strategies and update the supply of OUSD.
+     *      strategies and update the supply of OUSD, optionaly sending a
+     *      portion of the yield to the beneficiary.
      * @return uint256 New total supply of OUSD
      */
-    function _rebase()
-        internal
-        whenNotRebasePaused
-        returns (uint256 newTotalSupply)
-    {
-        if (oUSD.totalSupply() == 0) return 0;
-        uint256 oldTotalSupply = oUSD.totalSupply();
-        newTotalSupply = _totalValue();
-        // Only rachet upwards
-        if (newTotalSupply > oldTotalSupply) {
-            oUSD.changeSupply(newTotalSupply);
+    function _rebase() internal whenNotRebasePaused {
+        uint256 ousdSupply = oUSD.totalSupply();
+        if (ousdSupply == 0) {
+            return;
+        }
+        uint256 vaultSupply = _totalValue();
+
+        // Yield fee collection
+        address _beneficiaryAddress = beneficiaryAddress; // gas savings
+        if (_beneficiaryAddress != address(0) && (vaultSupply > ousdSupply)) {
+            uint256 yield = vaultSupply - ousdSupply;
```

Comment:
SafeMath :)

Reply:
I did the check in the "if" in the line before, but defense in depth, and being resilient to future code changes for the win! Definitely should use be using it here. Thanks for spotting. Fixed.